projects/step-frontend/src/lib/modules/timeseries/analytics/analytics-page.component.ts

```diff
@@ -0,0 +1,167 @@
+    let start;
+    let end;
+    let urlParams = TsUtils.getURLParams(window.location.href);
+    if (urlParams.refresh === '1') {
+      this.refreshEnabled = true;
+      this.selectedRefreshInterval = this.refreshIntervals[0];
+    }
+    if (urlParams.start) {
+      start = parseInt(urlParams.start);
+      end = parseInt(urlParams.end) ? parseInt(urlParams.end) : now;
+      this.timeRangeSelection = { type: RangeSelectionType.ABSOLUTE, absoluteSelection: { from: start, to: end } };
+    } else {
+      start = now - selectedTimeRange.timeInMs;
+      end = now;
+      this.timeRangeSelection = { type: RangeSelectionType.RELATIVE, relativeSelection: this.timeRangeOptions[0] };
+    }
+
+    delete urlParams.refresh; // in case it exists
+    delete urlParams.start;
+    delete urlParams.end;
+
+    this.dashboardSettings = {
+      contextId: new Date().getTime().toString(),
+      includeThreadGroupChart: true,
+      timeRange: { from: start, to: end },
+      contextualFilters: urlParams,
+      filterOptions: [
+        {
+          label: 'Status',
+          attributeName: 'rnStatus',
+          type: FilterBarItemType.OPTIONS,
+          textValues: [
+            { value: 'PASSED' },
+            { value: 'FAILED' },
+            { value: 'TECHNICAL_ERROR' },
+            { value: 'INTERRUPTED' },
+          ],
+          isLocked: true,
+        },
+        {
+          label: 'Type',
+          attributeName: 'type',
+          type: FilterBarItemType.OPTIONS,
+          textValues: [{ value: 'keyword' }, { value: 'custom' }],
+          isLocked: true,
+        },
+        {
+          label: 'Name',
+          attributeName: 'name',
+          type: FilterBarItemType.FREE_TEXT,
+          isLocked: true,
+        },
+        {
+          label: 'Execution Id',
+          attributeName: 'eId',
+          type: FilterBarItemType.FREE_TEXT,
+          isLocked: true,
+        },
+        {
+          label: 'Origin',
+          attributeName: 'origin',
+          type: FilterBarItemType.FREE_TEXT,
+          isLocked: true,
+        },
+        {
+          label: 'Plan Id',
+          attributeName: 'planId',
+          type: FilterBarItemType.FREE_TEXT,
+          isLocked: true,
+        },
+      ],
+    };
+    if (this.refreshEnabled) {
+      this.startInterval(this.selectedRefreshInterval.value);
+    }
+  }
+
+  changeRefreshInterval(newInterval: { label: string; value: number }) {
+    const oldInterval = this.selectedRefreshInterval;
+    this.selectedRefreshInterval = newInterval;
+    if (oldInterval.value === newInterval.value) {
+      return;
+    }
+    this.stopInterval$.next(null);
+    if (newInterval.value) {
+      this.startInterval();
+    }
+  }
+
+  startInterval(delay = 0) {
+    this.stopInterval$.next(null); // make sure to stop it if it's still running
+    timer(delay, this.selectedRefreshInterval.value)
+      .pipe(takeUntil(this.stopInterval$), takeUntil(this.terminator$))
```

Comment:
If you control your subscription by `stopInternval$`, `terminator$` is redundant
Just don't forget implement ngOnDestroy, emit value to stopInterval$, and complete it

Reply:
stopInterval is just for this particular interval, while terminator$ is the global destroy trigger. I think having both is better here.